### PR TITLE
Update rust-vmm-ci

### PR DESCRIPTION
**Pull request by roypat**
Signed-off-by: Patrick Roy <roypat@amazon.co.uk>

## Reason for This PR

The CI submodule hasn't been updated in three (!) years and is blocking any new contributions

## Description of Changes

Updates the CI submodule

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

**Comment by roypat** [comment]
rust-vmm-ci has changed from having a pipeline.yml file in their .buildkite directory to using a autogenerate_pipeline.py script. Thus the step `buildkite-agent pipeline upload rust-vmm-ci/.buildkite/pipeline.yml` fails.

It seems rust-vmm/linux-loader instead does `./rust-vmm-ci/.buildkite/autogenerate_pipeline.py | buildkite-agent pipeline upload`, so our buildkite config needs to be updated to accommodate this.

**Comment by roypat** [comment]
Because the code coverage message is so amusing, here's why it says our coverage increased from 33.3 to >90: The name of the files rust-vmm-ci expects to find to read code coverage information from got changed. The code trying to find the files is [here](https://github.com/rust-vmm/rust-vmm-ci/blob/7af2e16d9150869f252358a66adb2c5da1deec75/integration_tests/test_coverage.py#L16). There's a helpful comment there that tells us the script tries to avoid picking up the coverage files from the submodule, but if no other coverage data files exist, it will still do exactly that. And [that](https://github.com/rust-vmm/rust-vmm-ci/blob/main/coverage_config_x86_64.json) is where the 33.3 comes from

**Review by alxiord — COMMENTED** [review]
While you're at it :smile: could you also upgrade to Rust edition 2021 and fix the caret dependency issue? And maybe upgrade `vmm-sys-util` too.

**Comment by roypat** [comment]
> While you're at it 😄 could you also upgrade to Rust edition 2021 and fix the caret dependency issue? And maybe upgrade `vmm-sys-util` too.

That was the plan! The vmm-sys-util and dependency updates are in https://github.com/firecracker-microvm/micro-http/pull/31 (I decided to split this into two PRs since the rust-vmm-ci update turned out to involve more than a single commit), and jcawl@ has a PR for updating to rust 2021 that just needs to get rebased here https://github.com/firecracker-microvm/micro-http/pull/30

**Review by JonathanWoollett-Light — DISMISSED** [review]

**Review by zulinx86 — DISMISSED** [review]
nit: the message of the commit a34401ec238292067a9bed8ab7534221b4b18da1 looks broken.
```
Clear all clippy warnings
Additionally, HttpServer::new_from_fd has been marked as unsafe,
since the correctness of the unsafe call within the function
relies on the caller upholding an invarianClear all clippy warnings

Additionally, HttpServer::new_from_fd has been marked as unsafe,
since the correctness of the unsafe call within the function
relies on the caller upholding an invariant
```

**Comment by roypat** [comment]
> nit: the message of the commit [a34401e](https://github.com/firecracker-microvm/micro-http/commit/a34401ec238292067a9bed8ab7534221b4b18da1) looks broken.
> 
> ```
> Clear all clippy warnings
> Additionally, HttpServer::new_from_fd has been marked as unsafe,
> since the correctness of the unsafe call within the function
> relies on the caller upholding an invarianClear all clippy warnings
> 
> Additionally, HttpServer::new_from_fd has been marked as unsafe,
> since the correctness of the unsafe call within the function
> relies on the caller upholding an invariant
> ```

Yep, that would be another instance of me being bad at vim. Fixing! :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+# Unreleased
+
+## Added
+
+- Implemented `Eq` for `common::headers::Encoding`, `common::headers::MediaType`,
```
Comment (JonathanWoollett-Light) [inline CHANGELOG.md:5]: Nit: I think it would be better to include the full path e.g. `std::cmp::Eq`

Comment (roypat) [reply CHANGELOG.md:5]: `Eq` is in the prelude and so prevalent in rust that I do not think this clarification is needed

**Review by JonathanWoollett-Light — APPROVED** [review]

**Review by alxiord — APPROVED** [review]

**Pull request merged**